### PR TITLE
[Enhance] Fetch Customer's Item code based on Customer Name or Customer Group

### PR DESCRIPTION
**Pull request by Zlash65**
Issue:- https://github.com/frappe/erpnext/issues/13049

Item Master :- 
screen shot 2018-03-06 at 9 24 41 pm

customer-item-code

If customer_name is already set, then changing the customer_group for it will be prohibited.

### CODE REVIEW

**Comment by joezsweet** [comment]
Would be possible to do this on supplier side as well?

File: erpnext/stock/doctype/item/item.js [hunk 1]
```diff
@@ -170,6 +170,36 @@ frappe.ui.form.on('Item Reorder', {
...
+	var grid_row = cur_frm.fields_dict['customer_items'].grid.grid_rows_by_docname[row.name];
+
+	if (!row.customer_name) {
+		return false;
+	}
+
+	frappe.model.with_doc("Customer", row.customer_name, function() {
+		var customer = frappe.model.get_doc("Customer", row.customer_name);
+		row.customer_group = customer.customer_group;
+		grid_row.refresh_field("customer_group");
```
Comment (nabinhait) [inline erpnext/stock/doctype/item/item.js]: no need for `grid_row` obj here. You can use `set_value` function, it also refresh fields after setting the value.

File: erpnext/stock/get_item_details.py [hunk 2]
```diff
@@ -396,8 +396,16 @@ def validate_conversion_rate(args, meta):
...
+		out.customer_item_code = None
 		customer_item_code = item_doc.get("customer_items", {"customer_name": args.customer})
-		out.customer_item_code = customer_item_code[0].ref_code if customer_item_code else None
+
+		if customer_item_code:
+			out.customer_item_code = customer_item_code[0].ref_code
+		else:
+			customer_group = frappe.db.get_value("Customer", args.customer, "customer_group")
+			customer_group_item_code = item_doc.get("customer_items", {"customer_group": customer_group})
+			if customer_group_item_code and not customer_group_item_code[0].customer_name:
```
Comment (nabinhait) [inline erpnext/stock/get_item_details.py]: This `if` condition is not required.

File: erpnext/stock/doctype/item_customer_detail/item_customer_detail.json [hunk 3]
```diff
@@ -33,24 +38,62 @@
...
+   "no_copy": 0,
+   "options": "Customer Group",
+   "permlevel": 0,
+   "precision": "",
+   "print_hide": 0,
+   "print_hide_if_no_value": 0,
+   "read_only": 0,
+   "remember_last_selected_value": 0,
+   "report_hide": 0,
+   "reqd": 1,
```
Comment (nabinhait) [inline erpnext/stock/doctype/item_customer_detail/item_customer_detail.json]: It should not be mandatory

**Review by nabinhait — DISMISSED** [review]

**Pull request merged**